### PR TITLE
Added iOS Devices Module

### PR DESCRIPTION
**Pull request by tuxudo**
Collects information about iOS devices that have been connected to the
client. Also, pull request/issue number 1,000! 💃

### CODE REVIEW

File: app/modules/ios_devices/ios_devices_model.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        $this->rs['software_version'] = 0;
+        $this->rs['ios_id'] = '';
+        $this->rs['product_type'] = '';
+        $this->rs['region_info'] = '';
+        $this->rs['serial'] = '';
+        $this->rs['use_count'] = 0;
+        $this->rs['imei'] = '';
+        $this->rs['meid'] = '';
+
+        $this->serial = $serial;
```
Comment (bochoven) [inline app/modules/ios_devices/ios_devices_model.php]: I think this line can be removed

File: app/modules/ios_devices/ios_devices_model.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            // Delete previous entries
+            $this->deleteWhere('serial_number=?', $this->serial_number);
+
+            // Process incoming ios_devices.plist
+            $parser = new CFPropertyList();
+            $parser->parse($data, CFPropertyList::FORMAT_XML);
+            $plist = $parser->toArray();
+
+            foreach ($plist as $preffile) {
+                foreach (array('serial', 'prefpath', 'build_version', 'connected', 'device_class', 'family_id', 'firmware_version', 'firmware_version_string', 'ios_id', 'product_type', 'region_info', 'use_count', 'imei', 'meid', 'software_version') as $item) {
```
Comment (bochoven) [inline app/modules/ios_devices/ios_devices_model.php]: I think you should make this the outer loop and get rid of the other one

File: app/modules/ios_devices/scripts/ios_devices [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+#    if os.path.isfile('/Library/Preferences/com.apple.iPod.plist'):
+#        prefs = "/Library/Preferences/com.apple.iPod.plist" + "\n"
+#    else:
+#        prefs = ""
+#   Commented out because this file is deprecated^^
+
+    prefs = ""
+        
+    # Get user prefs
+    for file in glob.glob('/Users/*/Library/Preferences/com.apple.iPod.plist'):
```
Comment (bochoven) [inline app/modules/ios_devices/scripts/ios_devices]: User directories can be anywhere..

**Comment by ofirgalcon** [comment]
works. no prblems so far.

**Comment by bochoven** [comment]
Querying user home folders will end in 10.14

**Comment by tuxudo** [comment]
I don't think the ~/Library/Preferences folder is locked down like the mail/address book/messages folders. I tested with in Mojave (and found/fixed a bug) and it reported back results for all users when run as root.

**Comment by tuxudo** [comment]
Closing PR, moved to module repo

**Pull request closed without merging**